x-pack/legacy/plugins/index_management/public/components/mappings_editor/components/document_fields/fields/field_types/boolean_type.tsx

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React from 'react';
+
+import { i18n } from '@kbn/i18n';
+
+import { NormalizedField, Field as FieldType } from '../../../../types';
+import { getFieldConfig } from '../../../../lib';
+import { UseField, Field, FIELD_TYPES } from '../../../../shared_imports';
+import {
+  StoreParameter,
+  IndexParameter,
+  DocValuesParameter,
+  BoostParameter,
+  NullValueParameter,
+} from '../../field_parameters';
+import { EditFieldSection, AdvancedSettingsWrapper } from '../edit_field';
+
+const getDefaultValueToggle = (param: string, field: FieldType) => {
+  switch (param) {
+    case 'boost': {
+      return field[param] !== undefined && field[param] !== getFieldConfig(param).defaultValue;
+    }
+    case 'null_value': {
+      return field.null_value !== undefined && field.null_value !== '';
+    }
+    default:
+      return false;
+  }
+};
+
+interface Props {
+  field: NormalizedField;
+}
+
+const nullValueOptions = [
+  {
+    value: '"true"',
+    text: i18n.translate('xpack.idxMgmt.mappingsEditor.stringTrueFieldDescription', {
+      defaultMessage: '"true"',
+    }),
+  },
+  {
+    value: 'true',
+    text: i18n.translate('xpack.idxMgmt.mappingsEditor.booleanTrueFieldDescription', {
+      defaultMessage: 'true',
+    }),
+  },
+  {
+    value: '"false"',
+    text: i18n.translate('xpack.idxMgmt.mappingsEditor.stringFalseFieldDescription', {
+      defaultMessage: '"false"',
+    }),
+  },
+  {
+    value: 'false',
+    text: i18n.translate('xpack.idxMgmt.mappingsEditor.booleanFalseFieldDescription', {
+      defaultMessage: 'false',
+    }),
+  },
+];
+
+export const BooleanType = ({ field }: Props) => {
+  return (
+    <>
+      <EditFieldSection>
+        <StoreParameter />
+        <IndexParameter hasIndexOptions={false} />
+        <DocValuesParameter />
+      </EditFieldSection>
+
+      <AdvancedSettingsWrapper>
+        <EditFieldSection>
+          {/* boost */}
+          <BoostParameter defaultToggleValue={getDefaultValueToggle('boost', field.source)} />
+
+          {/* null_value */}
+          <NullValueParameter
+            defaultToggleValue={getDefaultValueToggle('null_value', field.source)}
+            description={i18n.translate(
+              'xpack.idxMgmt.mappingsEditor.booleanNullValueFieldDescription',
+              {
+                defaultMessage: 'Whether to substitute values for any explicit null values.',
+              }
+            )}
+          >
+            <UseField
```

Comment:
We can simplify a bit the `UseField` component with this logic:

```js
// above the component declare the const
const mapIndexToValue = ['true', true, 'false', false];

// Then each of the option has an index value (0, 1, 2, 3)

// in the jsx
<UseField
  path="null_value"
  config={{
    deserializer: (value: string | boolean) => mapIndexToValue.indexOf(value),
    serializer: (value: number) => mapIndexToValue[value],
  }}
  component={SelectField} // import directly the select from the "shared_imports"
  componentProps={{
    euiFieldProps: {
      options: nullValueOptions,
      style: { maxWidth: 300 },
    },
  }}
/>

```

What do you think?

Reply:
much cleaner, thanks for the suggestion!